horreum-web/src/auth/oidc.ts

```diff
@@ -0,0 +1,19 @@
+import {KeycloakConfig} from "../generated";
+import {UserManager, WebStorageStateStore} from "oidc-client-ts";
+
+export const createUserManager = (config: KeycloakConfig): UserManager => {
+    return new UserManager({
+        authority: config.url + "/realms/horreum",
+        client_id: config.clientId ?? "",
+        redirect_uri: `${window.location.origin}/silent-check-sso`,
```

Comment:
this is not correct. this value is for the `silent_redirect_uri` property.

`redirect_uri` it should be 
```
redirect_uri: `${window.location.origin}/${window.location.pathname}`
```
(assuming the OIDC provider accepts all redirects, which I believe it does). if not the alternative is to track the pathname in local state

the idea here is when the user press login from some page (say a test or a dataset page) it gets redirected to that same page after login.

Reply:
I am still banging my head trying to make it work the proper way but using 
```
redirect_uri: `${window.location.origin}/${window.location.pathname}`
// or
redirect_uri: `${window.location.origin}${window.location.pathname}`
```

does not work, the authentication itself works and it redirects back but.. for some reasons the context does not recognize the authentication to be succeeded